### PR TITLE
[1LP][WIPTEST] Enhancements to dockerbot setup.sh for easier debugging

### PR DESCRIPTION
**Pull request by bsquizz**
Fixes PR tester after recent base_url changes, since appliances 'hostname' must have a string.

### CODE REVIEW

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

File: cfme/test_framework/appliance.py [hunk 1]
```diff
@@ -44,11 +44,11 @@ def appliances_from_cli(cli_appliances):
 def pytest_configure(config):
 
     reporter = terminalreporter.reporter()
-    if stack.top:
-        appliances = [stack.top]
-    elif config.getoption('--dummy-appliance'):
+    if config.getoption('--dummy-appliance'):
```
Comment (RonnyPfannschmidt) [inline cfme/test_framework/appliance.py]: looks good at first glance, we still might want to know why stack.top is set here, i suspect a deeper issue and would love to see  the root cause in future

File: cfme/utils/dockerbot/pytestbase/setup.sh [hunk 2]
```diff
@@ -183,13 +183,23 @@ if [ -n "$PROVIDER" ]; then
...
+if [ "$USE_SPROUT" == "no" ];
+    HOSTNAME="$IP_ADDRESS"
+else
+    HOSTNAME="SPROUT_SHOULD_OVERRIDE_THIS"
+fi    
+
+cat >> $CFME_REPO_DIR/conf/env.local.yaml <<EOF
 appliances:
-  - hostname: $IP_ADDRESS
+  - hostname: $HOSTNAME
```
Comment (RonnyPfannschmidt) [inline cfme/utils/dockerbot/pytestbase/setup.sh]: adding on this we can actually grep for SPROUT_SHOULD_OVERRIDE_THIS in the sprout test and bail out correctly

Comment (bsquizz) [reply cfme/utils/dockerbot/pytestbase/setup.sh]: Good point, see latest commits

File: cfme/utils/dockerbot/pytestbase/setup.sh [hunk 3]
```diff
@@ -57,6 +58,9 @@ do_or_die () {
             sleep "$sleep_duration"
         else
             log "Failed to run the command $try times - exiting now..."
+	    if [ "$4" ]; then
```
Comment (RonnyPfannschmidt) [inline cfme/utils/dockerbot/pytestbase/setup.sh]: 👍

**Pull request merged**